static/src/javascripts/projects/common/modules/atoms/youtube.js

```diff
@@ -0,0 +1,241 @@
+    );
+    const playerHeight = player ? player.offsetHeight : 0;
+    const headline = document.querySelector(
+        '.immersive-main-media__headline-container'
+    );
+    const headlineHeight = headline ? headline.offsetHeight : 0;
+    const buttonOffset = playerHeight - headlineHeight;
+    const immersiveInterface = document.querySelector(
+        '.youtube-media-atom__immersive-interface'
+    );
+
+    if (immersiveInterface) {
+        immersiveInterface.style.top = `${buttonOffset}px`;
+    }
+};
+
+const onPlayerReady = (atomId, overlay, iframe, event) => {
+    players[atomId] = {
+        player: event.target,
+        pendingTrackingCalls: [25, 50, 75],
+        iframe,
+    };
+
+    if (shouldAutoplay(atomId)) {
+        event.target.playVideo();
+    }
+
+    if (overlay) {
+        players[atomId].overlay = overlay;
+
+        if (
+            !!config.page.section &&
+            detect.isBreakpoint({
+                min: 'desktop',
+            })
+        ) {
+            players[atomId].endSlate = getEndSlate(overlay);
+        }
+    }
+
+    if (iframe && iframe.closest('.immersive-main-media__media')) {
+        updateImmersiveButtonPos();
+        window.addEventListener(
+            'resize',
+            debounce(updateImmersiveButtonPos.bind(null), 200)
+        );
+    }
+};
+
+const onPlayerStateChange = (atomId, event): void =>
+    Object.keys(STATES).forEach(checkState.bind(null, atomId, event.data));
+
+const checkElemForVideo = elem =>
+    fastdom.read(() => {
+        $('.youtube-media-atom', elem).each((el, index) => {
+            const iframe = el.querySelector('iframe');
+
+            if (!iframe) {
+                return;
+            }
+
+            // append index of atom as iframe.id must be unique
+            iframe.id += `/${index}`;
+
+            // append index of atom as atomId must be unique
+            const atomId = `${el.getAttribute('data-media-atom-id')}/${index}`;
+            // need data attribute with index for unique lookup
+            el.setAttribute('data-unique-atom-id', atomId);
+            const overlay = el.querySelector('.youtube-media-atom__overlay');
+
+            tracking.init(getTrackingId(atomId));
+
+            initYoutubePlayer(
+                iframe,
+                {
+                    onPlayerReady: onPlayerReady.bind(
+                        null,
+                        atomId,
+                        overlay,
+                        iframe
+                    ),
+                    onPlayerStateChange: onPlayerStateChange.bind(null, atomId),
+                },
+                iframe.id
+            );
+        });
+    });
+
+const checkElemsForVideos = (elems: ?Array<HTMLElement>) => {
+    if (elems && elems.length) {
+        elems.forEach(checkElemForVideo);
+    } else {
+        checkElemForVideo(document.body);
+    }
+};
+
+export { checkElemsForVideos, onVideoContainerNavigation };
```

Comment:
`checkElemsForVideos` ...where are we using this? The uses probably need to be updated :)